### PR TITLE
Rename pageTemplates config to customTemplates in theme.json

### PR DESCRIPTION
**Pull request by youknowriad**
Addresses comments in https://github.com/WordPress/gutenberg/pull/27778#pullrequestreview-584290384

 - Uses the right hook (theme_templates)
 - Renames the key to avoid to "customTemplates" in theme.json

**Testing instructions**

 -  Add custom files like `about.html`/ `custom.html` ... to the `block-templates` folder.
 - Add a `customTemplates` property to the root level of `experimental-theme.json` like so

```
	"customTemplates": {
		"custom": "My Custom Template",
                "postTypes": ["page", "post", "mycpt"]
	}
``` 

Makes sure the the templates show up on the "page attributes" panel of the editor sidebar.

### CODE REVIEW

**Comment by aristath** [comment]
Every time I see "page templates" or "custom templates" referenced I need to spend a few seconds figuring out what we mean in the context. The problem is that we have lots of things called "templates", is there any way we can make it a bit easier to understand that this refers to custom templates specifically for pages?

**Comment by youknowriad** [comment]
This is not specifically for "pages", it can work with any custom post type.

**Comment by mtias** [comment]
One thing I think we need to reconsider is where we declare templates and whether the template metadata could live in the template itself (as `core/template` attributes) or on its own separate files / calls (like block.json does). It seems a bit odd to use theme.json for this — we don't register patterns nor blocks in theme.json, and templates are the same in that regard.

_This feedback should not block this PR._

**Review by oandregal — APPROVED** [review]
Added some minimal docs to this. Verified this works fine.

File: docs/designers-developers/developers/themes/theme-json.md [hunk 1]
```diff
@@ -476,6 +477,24 @@ These are the current typography properties supported by blocks:
 
 [1] The heading block represents 6 distinct HTML elements: H1-H6. It comes with selectors to target each individual element (ex: core/heading/h1 for H1, etc).
 
+
+### Other theme metadata
```
Comment (youknowriad) [inline docs/designers-developers/developers/themes/theme-json.md:481]: Thanks for this @nosolosw

**Comment by nosolosw** [comment]
My understanding is that the `title` key is mandatory by the `postTypes` is not. As a follow-up to this, does it make sense to add some validation / escaping / pre-load for this fields as we have for `settings`/`styles`? Some potential ideas:

- only save the custom templates that have a title => `get_custom_templates` won't list them.
- pre-load the `postTypes` field with `page` if the theme doesn't define any (this way we don't need to replicate that page logic anywhere else and can be expressed in tests).

**Comment by youknowriad** [comment]
Makes a lot of sense. 

That said, as said by @mtias we might explore finding ways to define these inline in the files themselves in the future. I guess validation would still be useful but implemented differently in that case.

**Comment by nosolosw** [comment]
Created https://github.com/WordPress/theme-experiments/pull/202 for updating TT1-blocks.

**Pull request merged**